lms/djangoapps/lti_provider/tests/test_outcomes.py

```diff
@@ -9,10 +9,39 @@
 
 from lti_provider.models import GradedAssignment, LtiConsumer, OutcomeService
 import lti_provider.outcomes as outcomes
-import lti_provider.tasks as tasks
 from opaque_keys.edx.locator import CourseLocator, BlockUsageLocator
 
 
+def create_descriptor_mock(parent, course_key, block_id):
+    """
+    Build a mock object for a content descriptor. This method assumes that
+    parent descriptors are created before child descriptors in order to properly
+    wire up the parent and child accessors.
+    """
+    desc = MagicMock()
+    desc.location = BlockUsageLocator(
+        course_key=course_key,
+        block_type='problem',
+        block_id=block_id,
+    )
+    desc.children = []
+    desc.get_children = MagicMock(return_value=desc.children)
+    desc.get_parent = MagicMock(return_value=parent)
+    if parent:
+        parent.children.append(desc)
+    return desc
```

Comment:
Can you use `ItemFactory` instead and create the course tree? Given the kinds of traversals that are being done, I'd really like to see tests that use `check_mongo_calls()` to make sure we don't have serious performance surprises.